pyinterprod/interpro/signature.py

```diff
@@ -737,9 +762,99 @@ def update_citation(cur: cx_Oracle.Cursor, pmid: int) -> Optional[str]:
             )
             RETURNING PUB_ID INTO :11
             """,
-            (*citation, pub_id)
+            (*citation, pub_id),
         )
         return pub_id.getvalue()[0]
     else:
         logger.warning(f"Citation related to PMID {pmid} not found.")
         return None
+
+
+def update_citations(cur: cx_Oracle.Cursor):
+    cur.execute(
+        """
+        DELETE FROM INTERPRO.CITATION
+        WHERE PUB_ID NOT IN 
+        (
+            SELECT PUB_ID FROM INTERPRO.ENTRY2PUB
+            UNION
+            SELECT PUB_ID FROM INTERPRO.METHOD2PUB
+            UNION
+            SELECT PUB_ID FROM INTERPRO.SUPPLEMENTARY_REF
+        )
+        """
+    )
+
+    pmids = _get_used_citations_pmids(cur)
+    step = 1000
+    for i in range(0, len(pmids), step):
+        params = list(pmids)[i: i + step]
```

Comment:
I made a suggestion below for `_get_used_citations_pmids()` to return a list instead of a set, so:

```suggestion
    pmids = _get_used_citations_pmids(cur)
    step = 1000
    for i in range(0, len(pmids), step):
        params = pmids[i:i+step]
```